### PR TITLE
fix multi recipient royalties on Details Tab

### LINKED ISSUE #6970: multi recipient royalties on Details Tab
### What happened?

Screenshot_20230826-154522_Firefox

https://kodadot.xyz/ahk/collection/101

### Please reproduce in steps

Above link

### Expected Behavior

To display multi royalties in a human readable fashion

### What browsers are you seeing the problem on?

_No response_

### At which address did you encounter bug?

_No response_

### Are you logged in?

None

### Which wallet you are using?

_No response_

### Screenshots

Screenshot_20230826-154522_Firefox

### Relevant log output

_No response_

### Payment link for reward

https://kodadot.xyz/ksm/transfer/?target=DT7kRjGFvRKxGSx5CPUCA1pazj6gzJ6Db11xmkX4yYSNK7m

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by niklasp**
## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #6970 
- [ ] Requires deployment <snek/rubick/worker>

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1soESeTVLfse9e2G8dRSMUyJ2SWad33qhtkjQTv9GMToRvP

### <samp>🤖 Generated by Copilot at 7483263</samp>

Display and style NFT transfer recipients in `GalleryItemDescription.vue`. Use a computed property to format the data and a template to render it.

### <samp>🤖 Generated by Copilot at 7483263</samp>

> _Oh we're the coders of the NFT sea_
> _And we display the data as it should be_
> _With a computed property and a template so fine_
> _And some custom CSS classes to make it shine_

## Description

backward compatible version of multiroyalties according to the design guide given

http://localhost:9090/ahk/gallery/101-3962765076 => multi royalty) in array format [[addr0, perc0], [addr1, perc1], ...]
Bildschirmfoto 2023-08-28 um 19 14 29

http://localhost:9090/ahk/gallery/101-3163187696 => single royalty in array format [[addr0,perc0]]
Bildschirmfoto 2023-08-28 um 19 14 33

Bildschirmfoto 2023-08-28 um 19 14 36

http://localhost:9090/ksm/gallery/19381324-BC4ECCF7AB22138D66-SMS-SOL_SMS_22_KSM-00000001 (old royalty, just address)

### CODE REVIEW

File: components/gallery/GalleryItemDescription.vue [hunk 1]
```diff
@@ -257,3 +288,25 @@ const openLink = (link) => {
   window.open(toCloudflareIpfsUrl(link), '_blank')
 }
 </script>
+
+<style lang="scss">
+@import '../../styles/abstracts/variables.scss';
+.recipient {
+  text-transform: capitalize;
```
Comment (roiLeo) [inline components/gallery/GalleryItemDescription.vue:295]: [Typographie helpers](https://bulma.io/documentation/helpers/typography-helpers/) might help

File: components/gallery/GalleryItemDescription.vue [hunk 2]
```diff
@@ -257,3 +288,25 @@ const openLink = (link) => {
...
+
+  > ul > li {
+    display: flex;
+    flex-direction: row;
+    gap: 0.5rem;
+    align-items: center;
+    justify-content: flex-end;
```
Comment (roiLeo) [inline components/gallery/GalleryItemDescription.vue:302]: [Flexbox helpers](https://bulma.io/documentation/helpers/flexbox-helpers/)

File: components/gallery/GalleryItemDescription.vue [hunk 3]
```diff
@@ -85,14 +85,35 @@
...
-        <nuxt-link
-          :to="`/${urlPrefix}/u/${nft?.recipient}`"
-          class="has-text-link">
-          <Identity ref="identity" :address="nft?.recipient" />
-        </nuxt-link>
+        v-if="recipient"
+        class="recipient is-flex is-justify-content-space-between">
+        <template v-if="Array.isArray(recipient) && recipient.length > 1">
+          <p>{{ $t('transfers.recipients') }}</p>
+          <ul>
```
Comment (roiLeo) [inline components/gallery/GalleryItemDescription.vue:92]: You could have use `<ol>` tag here to display numbers

File: components/gallery/GalleryItemDescription.vue [hunk 4]
```diff
@@ -85,14 +85,35 @@
...
+          v-else-if="Array.isArray(recipient) && recipient.length === 1">
+          <p>{{ $t('transfers.recipient') }}</p>
+          <nuxt-link
+            :to="`/${urlPrefix}/u/${recipient[0][0]}`"
+            class="has-text-link">
+            <Identity ref="identity" :address="recipient[0][0]" />
+          </nuxt-link>
+        </template>
+        <template v-else>
+          <p>{{ $t('transfers.recipient') }}</p>
```
Comment (roiLeo) [inline components/gallery/GalleryItemDescription.vue:112]: I have the impression that this line is repeated more than once, maybe put it on top/outside if-else

File: components/gallery/GalleryItemDescription.vue [hunk 5]
```diff
@@ -257,3 +288,25 @@ const openLink = (link) => {
   window.open(toCloudflareIpfsUrl(link), '_blank')
 }
 </script>
+
+<style lang="scss">
+@import '../../styles/abstracts/variables.scss';
```
Comment (roiLeo) [inline components/gallery/GalleryItemDescription.vue:293]:
```suggestion
@import '@/styles/abstracts/variables.scss';
```

File: components/gallery/GalleryItemDescription.vue (hunk 3 above)
Comment (niklasp) [reply components/gallery/GalleryItemDescription.vue:92]: there was an issue with displaying numbers and flex li items i kept it

File: components/gallery/GalleryItemDescription.vue (hunk 4 above)
Comment (niklasp) [reply components/gallery/GalleryItemDescription.vue:112]: not a vue dev, how could this be optimized? all are slightly different here

**Comment by niklasp** [comment]
Thanks for the review. is it okay to not use a PR template, ill do next time i PR something, promise 🤞

File: components/gallery/GalleryItemDescription.vue (hunk 3 above)
Comment (roiLeo) [reply components/gallery/GalleryItemDescription.vue:92]: please can you try to fix it?

File: components/gallery/GalleryItemDescription.vue [hunk 6]
```diff
@@ -85,14 +85,38 @@
...
-        <p>{{ $t('transfers.recipient') }}</p>
-        <nuxt-link
-          :to="`/${urlPrefix}/u/${nft?.recipient}`"
-          class="has-text-link">
-          <Identity ref="identity" :address="nft?.recipient" />
-        </nuxt-link>
+        v-if="recipient"
+        class="recipient is-flex is-justify-content-space-between is-capitalized">
+        <template v-if="Array.isArray(recipient) && recipient.length > 1">
+          <p>{{ $t('transfers.recipients') }}</p>
```
Comment (roiLeo) [inline components/gallery/GalleryItemDescription.vue:91]:
```suggestion
        class="recipient is-flex is-justify-content-space-between is-capitalized">
        <p>{{ $t('transfers.recipients') }}</p>
        <template v-if="Array.isArray(recipient) && recipient.length > 1">
```

File: components/gallery/GalleryItemDescription.vue [hunk 7]
```diff
@@ -85,14 +85,38 @@
...
+              <nuxt-link :to="`/${urlPrefix}/u/${addr}`" class="has-text-link">
+                <Identity ref="identity" :address="addr" />
+              </nuxt-link>
+              <span className="is-size-7">({{ percentile }}%)</span>
+            </li>
+          </ul>
+        </template>
+        <template
+          v-else-if="Array.isArray(recipient) && recipient.length === 1">
+          <p>{{ $t('transfers.recipient') }}</p>
```
Comment (roiLeo) [inline components/gallery/GalleryItemDescription.vue:107]:
```suggestion
```

File: components/gallery/GalleryItemDescription.vue [hunk 8]
```diff
@@ -85,14 +85,38 @@
...
+          <nuxt-link
+            :to="`/${urlPrefix}/u/${recipient[0][0]}`"
+            class="has-text-link">
+            <Identity ref="identity" :address="recipient[0][0]" />
+          </nuxt-link>
+        </template>
+        <template v-else>
+          <p>{{ $t('transfers.recipient') }}</p>
```
Comment (roiLeo) [inline components/gallery/GalleryItemDescription.vue:115]:
```suggestion
```

File: components/gallery/GalleryItemDescription.vue (hunk 3 above)
Comment (niklasp) [reply components/gallery/GalleryItemDescription.vue:92]: you mean by using `<ol>` instead of `<ul>` with the index?

**Comment by niklasp** [comment]
@yangwao used template
@roiLeo merged suggestions and fixed ol issue

**Comment by niklasp** [comment]
@prury @preschian ready for review

**Review by roiLeo — APPROVED** [review]

**Comment by yangwao** [comment]
pay 50 usd
can go for next release

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 4.58 USD/DOT ~ 10.917 $DOT
🧗 [1soESeTVLfse9e2G8dRSMUyJ2SWad33qhtkjQTv9GMToRvP](https://polkadot.subscan.io/account/1soESeTVLfse9e2G8dRSMUyJ2SWad33qhtkjQTv9GMToRvP)
🔗 [0x86eaab8542e54a095fc96794c65c4cce3218f1e851dd0c2834efc56689b2f7bc](https://polkadot.subscan.io/extrinsic/0x86eaab8542e54a095fc96794c65c4cce3218f1e851dd0c2834efc56689b2f7bc)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by prury** [comment]
> pay 50 usd can go for next release

cool, before that we have to add fields on the NFT creation page to add more than one royalty recipient and percentages

**Comment by niklasp** [comment]
thanks for the assistance

**Pull request merged**